### DISCUSSION TITLE
Router push with URL object doesn't replace path parameter by the query property value

### DISCUSSION THREAD

**Discussion by souzantero** [vercel/next.js#18410] [Help] [answered]
Hi, I opened a new issue described on this link https://github.com/vercel/next.js/issues/18409. Does anyone know tell me if it is a real bug or a development fail? Thank you!

**Comment by jamesmosier** [comment, marked as answer]
As mentioned in this comment https://github.com/vercel/next.js/issues/18409#issuecomment-718195725 , combining hardcoded values with interpolation will not work:

This will not work:
```js
    router.push({
      pathname: 'posts/[postId]/comments/5',
      query: { postId: 1 }
    })
```

This will work:
```js
    router.push({
      pathname: 'posts/[postId]/comments/[commentId]',
      query: { postId: 1, commentId: 5 }
    })
```

or this will work:

```js
    router.push({
      pathname: 'posts/1/comments/5',
    })
```

but you cannot have a `pathname` with both `[some_value]` and `5`  (or any parameter) in the `pathname`
